src/main/groovy/life/qbic/controller/LocationsController.groovy

```diff
@@ -64,24 +71,25 @@ class LocationsController {
     }
   }
 
-  @Get(uri = "/{contact_email}", produces = MediaType.APPLICATION_JSON)
-  @Operation(summary = "Provides the locations information linked to an e-mail",
-          description = "Provides detailed locations information that is linked to an e-mail",
+  @Get(uri = "/{user_id}", produces = MediaType.APPLICATION_JSON)
+  @Operation(summary = "Provides the locations information linked to a username",
+          description = "Provides detailed locations information that is linked to a username",
           tags = "Contact")
-  @ApiResponse(responseCode = "200", description = "Current locations associated with the email address",
+  @ApiResponse(responseCode = "200", description = "Current locations associated with the username",
           content = @Content(
                   mediaType = "application/json",
                   schema = @Schema(implementation = Location.class)))
-  @ApiResponse(responseCode = "400", description = "Invalid e-mail address")
+  @ApiResponse(responseCode = "400", description = "Invalid user")
   @ApiResponse(responseCode = "401", description = "Unauthorized access")
   @ApiResponse(responseCode = "404", description = "Location not found")
   @RolesAllowed(["READER", "WRITER"])
-  HttpResponse<List<Location>> locations(@PathVariable('contact_email') String contact_email){
-    if(!RegExValidator.isValidMail(contact_email)) {
-      HttpResponse<Contact> res = HttpResponse.badRequest("Not a valid email address!")
+  HttpResponse<List<Location>> locations(@PathVariable('user_id') String userId){
+    // We take the assumption here that the user id is an email address. While this is true as of 1.1.0 we need to be careful here.
+    if(!RegExValidator.isValidMail(userId)) {
```

Comment:
Should we keep this test? The user ID is an email at the moment, but we don't explicitly state that nor does it has to be in my opinion.

Reply:
ok let's remove it :+1: